kong/runloop/balancer.lua

```diff
@@ -264,21 +265,24 @@ do
       -- The lifetime of the healthchecker is based on that of the balancer.
       healthcheckers[balancer] = hc
 
-      balancer.report_http_status = function(ip, port, status)
+      balancer.report_http_status = function(handle, status)
+        local ip, port = handle.address.ip, handle.address.port
```

Comment:
This line is returning an error on every request when I test this change combined with [fix for issue 49](https://github.com/Kong/lua-resty-dns-client/issues/49) over Kong CE 1.0.3.

```
2019/03/08 13:59:40 [error] 5245#0: *166 failed to run log_by_lua*: /usr/local/share/lua/5.1/kong/runloop/balancer.lua:269: attempt to index field 'address' (a nil value)
stack traceback:
        /usr/local/share/lua/5.1/kong/runloop/balancer.lua:269: in function 'report_http_status'
        /usr/local/share/lua/5.1/kong/runloop/handler.lua:952: in function 'after'
        /usr/local/share/lua/5.1/kong/init.lua:772: in function 'log'
        log_by_lua(nginx-kong.conf:125):2: in function <log_by_lua(nginx-kong.conf:125):1> while logging request, client: 192.168.121.1, server: kong, request: "GET /v1/f174909b-40f1-4d71-ae5f-4fa86702b4ab.json HTTP/1.1", upstream: "http://52.95.164.33:80/v1/f174909b-40f1-4d71-ae5f-4fa86702b4ab.json", host: "192.168.121.46:8000"
```

Complete log available here: https://drive.google.com/file/d/1248kdT5u6oK0N5_bxUH1ZaPxu-kmyAC3/view?usp=sharing

For this test I just set up the server as described on my [comment for issue 49](https://github.com/Kong/lua-resty-dns-client/issues/49#issuecomment-463680659), and sent 10 requests (with a 1sec sleep between then), nothing in parallel. There was no error 500, but this also seems a bit weird in the logs.

Reply:
thx for testing, but you we're too early. I rebased this PR, but didn't get to fix the test failures yet. Sorry for non mentioning that.

Reply:
Sure, I understand. Just trying to help, since this fix is a big deal for us here.